### ISSUE TITLE
502 Bad Gateway in AWX when having Wappalyzer enabled in Chrome

### ISSUE THREAD

**Issue by maksimluzik** [AliasIO/Wappalyzer#1870]
I get a 502 Bad Gateway errors in my network tab and cant use Open Source Ansible Tower (AWX) at all. See attached screenshot:
![image](https://user-images.githubusercontent.com/7724063/33366187-28bb0592-d4f3-11e7-8121-2437b5afe506.png)

**Comment by AlexByte** [comment]
Have you tried disabling Wappalyzer?

**Comment by maksimluzik** [comment]
Yes, and disabling it removes this issue. But just wanted to report it here that that enabling disabling  addon is not convenient. And just don't understand in general, why it affects the AWX as it does.

**Comment by AliasIO** [comment]
I'm not sure what's happening here, it's a server-side error and Wappalyzer doesn't interfere with outgoing requests. The only thing I can think of is that Wappalyzer tries to access robots.txt in a separate request but I don't know why that would result in a 502 on the original request.

**Comment by OddEssay** [comment]
I've just had this exact same issue while running AWX inside a Vagrant VM, and on a second install running on a Hyper-V VM.

Since I was able to login when in another browser, or an Incognito mode I figure it _could_ be the same issue, disabled Wappalyzer, reloaded the page and it all works great.

Previously I'd tried rebooting the VM, deleting all local storage and cookies and cleaning all caches, hard refreshes of the page etc.

And if I re-enabled Wappalyzer, it stops working. Disable, starts working. 😢 

I tried comparing the requests to both `/api` (Which fails) and root request`/` (Which works) using "Copy as cURL" in the Chrome Dev tools Network Tab. But the results are character for character identical. (And work as expected when run from the command line)

**Comment by OddEssay** [comment]
TLDR: Wappalyzer hits `/robots.txt` and AWX throws a 502 for that request. If the next request is made within ~5 second it fails. This 2nd request seems to reset the web server and a 3rd request is fine. A wait of > 5 seconds seems enough for the web server to respond to the next request successfully.

It was bugging me why this would be an issue, so I was digging a little deeper into the  Wappalyzer options and noticed "Anonymously send reports on detected applications to wappalyzer.com for research"

Turns out, if I leave Wappalyzer enabled, but turn off that option everything with AWX works, but if I switch it on, the 502 errors come back.

Following with Wireshark, that is when it hits `robots.txt` as described above (Which doesn't appear in the Google Dev Tools Network Tab that I can see) and hitting that URL does return a 502 error normally.

After a few iterations I can reproduce the issue by using curl on the command line to hit both URLs in rapid succession using: `curl 'http://awx.local/robots.txt' -H 'Accept-Encoding: gzip, deflate' -H 'Accept-Language: en,en-GB;q=0.9' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36' -H 'Accept: application/json, text/plain, */*' -H 'Referer: http://awx.local/' -H 'Cookie: sessionExpired=false; lastPath=%2Fhome; userLoggedIn=false; current_user=%7B%7D' -H 'X-CookiesOK: I explicitly accept all cookies' -H 'Connection: keep-alive' --compressed; curl 'http://awx.local/api/' -H 'Accept-Encoding: gzip, deflate' -H 'Accept-Language: en,en-GB;q=0.9' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36' -H 'Accept: application/json, text/plain, */*' -H 'Referer: http://awx.local/' -H 'Cookie: sessionExpired=false; lastPath=%2Fhome; userLoggedIn=false; current_user=%7B%7D' -H 'X-CookiesOK: I explicitly accept all cookies' -H 'Connection: keep-alive' --compressed`

If I throw a `sleep 4;` in the middle, the second request still fails.
If I throw a `sleep 5;` in the middle, that seems long enough to clear it and the second request succeeds.
If I don't sleep at all, and add another request to `/api` that 3rd request successes consistently.

So, while Wappalyzer is triggering the error, it's not necessarily at fault or in a position to fix it.

**Comment by AliasIO** [comment]
Thanks for getting to the bottom of that one! I'll have a think about what can be done about it.
